test/base/resource_checks.go

```diff
@@ -44,18 +45,35 @@ const (
 // name the metric that is emitted to track how long it took for
 // the resource to get into the state checked by isResourceReady.
 func WaitForResourceReady(dynamicClient dynamic.Interface, obj *MetaResource) error {
-	metricName := fmt.Sprintf("WaitForResourceReady/%s", obj.Name)
+	metricName := fmt.Sprintf("WaitForResourceReady/%s/%s", obj.Namespace, obj.Name)
 	_, span := trace.StartSpan(context.Background(), metricName)
 	defer span.End()
 
 	return wait.PollImmediate(interval, timeout, func() (bool, error) {
-		return isResourceReady(dynamicClient, obj)
+		untyped, err := GetGenericObject(dynamicClient, obj, &duckv1beta1.KResource{})
+		return isResourceReady(untyped, err)
+	})
+}
+
+// WaitForResourcesReady waits until all the specified resources in the given namespace are ready.
+func WaitForResourcesReady(dynamicClient dynamic.Interface, objList *MetaResourceList) error {
+	metricName := fmt.Sprintf("WaitForResourcesReady/%s", objList.Namespace)
+	_, span := trace.StartSpan(context.Background(), metricName)
+	defer span.End()
+
+	return wait.PollImmediate(interval, timeout, func() (bool, error) {
+		untypeds, err := GetGenericObjectList(dynamicClient, objList, &duckv1beta1.KResource{})
+		for _, untyped := range untypeds {
+			if isReady, err := isResourceReady(untyped, err); !isReady {
```

Comment:
can it be isReady true and also returned an error? I guess I mean shouldn't you be checking for err != nil as well?

Reply:
It cannot, since it does not make sense if `isResourceReady` returns `the resource is ready but there is an error`.